src/boot/bootctl-uki.c

```diff
@@ -110,3 +112,77 @@ int verb_kernel_identify(int argc, char *argv[], void *userdata) {
         puts("unknown");
         return EXIT_SUCCESS;
 }
+
+static int read_pe_section(FILE *uki, const struct PeSectionHeader *section,
+                           void **ret, size_t *ret_n) {
+        _cleanup_free_ void *data = NULL;
+        uint32_t size, bytes;
+        uint64_t soff;
+        int rc;
+
+        soff = le32toh(section->PointerToRawData);
+        size = le32toh(section->VirtualSize);
+
+        rc = fseek(uki, soff, SEEK_SET);
+        if (rc < 0)
+                return log_error_errno(errno, "seek to PE section");
+
```

Comment:
hmm, we probably should add a safety check that size is not beyond all bounds before we allocate memory this large. i.e. check if the size is not larger than 16M or so?

Reply:
For the given use (inspect the small stuff) we can probably go with a much lower limit.